graphql_compiler/schema_generation/schema_graph.py

```diff
@@ -652,9 +653,20 @@ def _get_graphql_type(self, class_name, property_definition):
                                          'collection property {}. Only graph classes are allowed '
                                          'to have collections as properties.'
                                          .format(class_name, property_definition))
-                # Don't include the fields and implemented interfaces, this information is already
-                # stored in the SchemaGraph.
-                graphql_type = GraphQLList(GraphQLObjectType(linked_class, {}, []))
+                if len(self._inheritance_sets[linked_class]) > 1:
+                    raise AssertionError('Class {} contains an invalid collection of class {} '
+                                         'elements. Inner classes are not allowed to have '
+                                         'multiple superclasses.'.format(class_name, linked_class))
```

Comment:
I think this message is a bit confusing? Why is it specifically `>1` (rather than `0`)?

Reply:
Ok so each class is a superclass of itself. I will change the if condition to be more explicit about this in the next update.